modules/web-console/frontend/app/modules/agent/AgentManager.service.ts

```diff
@@ -425,7 +432,19 @@ export default class AgentManager {
                     latch.resolve(evt.payload);
             });
 
-        this._sendWebSocketEvent(requestId, eventType, data);
+        this.wsConnection.pipe(
+            filter((isConnected) => isConnected),
```

Comment:
On my previous project, I moved the wsConnection-filter-first into separate observable called `connectionReady` and used it to wait for it before making all requests. You might want to at least extract it too, this would indicate the significance of it and reuse in other methods, like `_sendWebSocketEvent` if we decide to.

Also, this observable chain introduces new behavior: agent requests now have an explicit 5s timeout while there was none before. This timeout is different from the option in Sockette options, which tells for how long to wait to reconnect after ECONNREFUSED error. I'd at least test all other screens for regressions.